### ISSUE TITLE
[TW-245] Dependencies: completing a task results in "Dependencies set to '0'"

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#288]
_Kosta Harlan on 2013-07-10T15:47:28Z says:_ 

 Not sure if this is desired behavior or not.

If I do the following:

> task add "Bake bread" (task 1 created)
> task add "Buy flour" (task 2 created)
> task 1 mod depends:2 (task 2 is now blocking task 1)
> task 2 done
> task 1 info (Dependencies set to '0')

If there are no more dependencies, shouldn't dependencies be set to NULL (the equivalent of "task 1 mod depends:")?

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2013-07-10T15:47:28Z
Modified: 2014-12-28T01:22:48Z
```

**Comment by taskwarrior** [comment]
_Kosta Harlan on 2013-07-11T16:47:57Z says:_ 

 Also if I run "task 1 export" after completing task 2, I see the uuid for task 2 in the "depends" section for task 1. Not sure if that is correct either since task 2 has been completed at that point.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2013-07-11T17:15:43Z says:_ 

 Completing a task does not remove it as a dependency.  The relationship is preserved deliberately.  To determine blocked/blocking status the dependency chain is followed.  This is as expected.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2013-09-29T18:02:54Z says:_ 

 I think this is just a problem with the feedback message.  The dependencies are preserved despite status, and dynamically determined at run time.

**Comment by taskwarrior** [comment]
_Marek Vitek on 2013-12-18T10:36:02Z says:_ 

 Hello Paul, thank you for great tool. 
It is not just matter of data output. It is problem with filtering dependent tasks. Let me present you scenario I am used to where it will be useful to remove dependency upon task completion.

<pre>
task add task1
task add task2 depend:1
</pre>
create following filter in your view:
report.workable.filter=status:pending depends.none: 

Now you can see only task1 in your workable list.
If you finish task1 I would expect task2 will appear on my list as it is not being blocked by any active task and I can work on it. But it is not case.
<pre>
$ task 1 done
</pre>
Now you can see no tasks in your workable report. Also task depends shows task 2 but dependency is empty.

Maybe there is other way around this limitation but I didn't found it. I have tried to use depends.count as it is set to [0] when parent task is completed but I wasn't able to make it work.

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-02-19T19:54:25Z says:_ 

 Marek you can use the special virtual tag -BLOCKED for that behavior.
It filters tasks based on the completion status of its dependencies.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-12-28T01:22:48Z says:_ 

 Cannot replicate - the 'task 1 info' output contains *no* reference to any dependencies.  This bug was reported against 2.2.0, so I suspect this was fixed ealier, in 2.3.0.
